### ISSUE TITLE
Kamikazes don't kill marines on board

### ISSUE THREAD

**Issue by simon33-2** [triplea-game/triplea#4634]
### Engine version
1.9-13066

### My Operating System
Ubuntu 16.04.3

### Map name
World War II Global 1940 Balanced Mod3

### Can you describe how to trigger the error? (eg: what sequence of actions will recreate it?)
Put a marine on board a US (normally) Battleship or Cruiser
Marine amphibiously assaults an empty Japanese territory in a Kamikaze zone.
Japan sinks the ship with Kamikazes

Marine who should be dead takes territory.

### Instead of this error, what should have happened?
Marine killed. Territory remains Japanese.

### Any additional information that may help
Refer #4162. Probably the same problem but you never know.

**Comment by simon33-2** [comment]
Seems that this is an old bug. Just tested on builds 11599, 7378 and 3627. All had the bug.

**Comment by panther2** [comment]
Does it still occur on versions above 12640?

**Comment by simon33-2** [comment]
Well it still occurs on 13066.

**Comment by simon33-2** [comment]
Test case.
[marine.zip](https://github.com/triplea-game/triplea/files/2871179/marine.zip)

**Comment by DanVanAtta** [comment]
@simon33-2 @panther2 , I notice in the rules for global: "Japan may target any surface warship in a Kamikaze zone"; I'm not a global player, is it 100% intentional that transports are not eligible for kamikaze attacks? 

It seems odd that a cruiser doing an amphib assault can be sunk via kamikaze but not a transport. I presume you left that out for good reason, just want to confirm that it's how it should be, that transports by themselves are not eligible for kamikaze attacks. Example (1.9.0.0.13066):
[marine2.tsvg.zip](https://github.com/triplea-game/triplea/files/4660376/marine2.tsvg.zip)

**Comment by Cernelius** [comment]
>"Japan may target any surface warship in a Kamikaze zone"; I'm not a global player, is it 100% intentional that transports are not eligible for kamikaze attacks?

Yes. Transports are ships, but they are not warships. What that rule is saying, is that you can target any ships but transports (non-war) and submarines (non-surface).

Not much of an idea why transports cannot be targeted at all, instead of being simply deprioritized, as usual.

**Comment by DanVanAtta** [comment]
Ya, just an odd inconsistency.

 The way transports are tracked is very weak. A fix here I think either
needs a gameData datastructure to replace or augment the various flags for
unit transport or we need to be able to look into history to check for
offloaded units. Either way perhaps not a quick fix to correct this.
